Feat/datadog tags (#1551)
Kong/docs.konghq.com: app/_hub/kong-inc/datadog/index.md

---

## Metrics

> Plugin currently logs following metrics to the Datadog server about a Service or Route.

Reviewer: ```suggestion
Plugin currently logs following the metrics to the Datadog server about a Service or Route.
```

Metric                     | description | namespace
---                        | ---         | ---
`request_count`            | tracks the request | kong.request.count
`request_size`             | tracks the request's body size in bytes | kong.request.size
`response_size`            | tracks the response's body size in bytes | kong.response.size
`latency`                  | tracks the time interval between the request started and response received from the upstream server | kong.latency
`upstream_latency`         | tracks the time it took for the final service to process the request | kong.upstream\_latency
`kong_latency`             | tracks the internal Kong latency that it took to run all the plugins | kong.kong\_latency

> The metrics will be send with the tags `name` and `status` carrying the API name and HTTP status code respectively. If you specifying `consumer_identifier` with the metric, a tag `consumer` be added too.

Reviewer: ```suggestion
The metrics will be sent with the tags `name` and `status` carrying the API name and HTTP status code respectively. If you specify `consumer_identifier` with the metric, a tag `consumer` be added too.
```

1.  by default all metrics get logged.
2.  metric with `stat_type` as `counter` or `gauge` must have `sample_rate` defined as well.

> ## Migrating datadog queries

Reviewer: ```suggestion
## Migrating DataDog queries
```

> With the plugin, the api, status and consumer specific metrics are replaced by a generic metric name.

Reviewer: ```suggestion
With the plugin, the api, status, and consumer-specific metrics are replaced by a generic metric name.
```

> You have to change your datadog queries in dashboard and alerts to reflect this:

Reviewer: ```suggestion
You have to change your DataDog queries in dashboard and alerts to reflect this:
```

> For example the following query:

Reviewer: ```suggestion
For example, the following query:
```
